### PR TITLE
[PLGN-71] Addition of SentinelOne service user

### PR DESCRIPTION
**Pull request by dlaverty-r7**
### Description
https://issues.corp.rapid7.com/browse/PLGN-71
SentinelOne now provides a 'Service user' role, which allows API only access to the console (no UI access). This ticket supports the addition of this role (which uses the API key directly in the header during calls) and also removed the Basic authentication, which is no longer supported. Note that the user_type allows distinguishing between the original user type 'console user' and the new 'service user'. 
Note it was also necessary to make a number of updates in the help.md file to ensure examples were provided correctly, in order to pass the validation steps in the make validate step. 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/sentinelone/komand_sentinelone/connection/connection.py [hunk 1]
```diff
@@ -112,37 +109,28 @@ def _handle_auth_response(self, url: str, response):
...
+            return {DATA_FIELD: {API_TOKEN_FIELD: self.api_key}}
+        else:
+            raise PluginException(
+                cause="Inputs related to API key authentication is invalid.",
+                assistance="Check API key input and try again. "
+                "If the problem persists contact with development team.",
+            )
 
     def make_token_header(self):
+        if self.role == 'Service role':
```
Comment (jkaluza-rapid7) [inline plugins/sentinelone/komand_sentinelone/connection/connection.py:125]: You could move 'Service role' to constants too :)

**Review by jkaluza-rapid7 — APPROVED** [review]
Great job! Especially at winning the battle with the validators

File: plugins/sentinelone/komand_sentinelone/connection/connection.py [hunk 2]
```diff
@@ -70,6 +65,8 @@ def _get_start_index(url):
 
     def get_auth_token(self) -> Tuple[str, str]:
         version = "2.1"
+        if self.role == 'Service role':
```
Comment (igorski-r7) [inline plugins/sentinelone/komand_sentinelone/connection/connection.py:68]: I suggest moving that `Service role` to a constant variable 😄

File: plugins/sentinelone/help.md [hunk 3]
```diff
@@ -644,18 +638,16 @@ Example input:
 ```
 {
   "agent": "hostname123",
-  "filter": {
-    "updatedAt__gt": "2019-02-27T04:49:26.257525Z"
-  },
+  "filter": "{ \"updatedAt__gt\": \"2019-02-27T04:49:26.257525Z\" }",
```
[the remark this answers is not in the document]
Comment (dlaverty-r7) [reply plugins/sentinelone/help.md]: Updated :)

File: plugins/sentinelone/unit_test/util.py [hunk 4]
```diff
@@ -25,8 +25,7 @@ def default_connector(action, connect_params: object = None):
         else:
             params = {
                 Input.URL: "https://rapid7.com",
-                Input.AUTHENTICATION_TYPE: "Basic Auth",
-                Input.BASIC_AUTH_CREDENTIALS: {"username": "username", "password": "password"},
+                Input.ROLE: "User role",
```
[the remark this answers is not in the document]
Comment (dlaverty-r7) [reply plugins/sentinelone/unit_test/util.py]: Updated

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**